x/genutil/client/cli/init.go

```diff
@@ -114,12 +118,47 @@ func InitCmd(mbm module.BasicManager, defaultNodeHome string) *cobra.Command {
 			genFile := config.GenesisFile()
 			overwrite, _ := cmd.Flags().GetBool(FlagOverwrite)
 
+			stakingBondDenom, _ := cmd.Flags().GetString(FlagStakingBondDenom)
+
 			if !overwrite && tmos.FileExists(genFile) {
 				return fmt.Errorf("genesis.json file already exists: %v", genFile)
 			}
-			appState, err := json.MarshalIndent(mbm.DefaultGenesis(cdc), "", " ")
+
+			var defaultGenesis map[string]json.RawMessage
+
+			if stakingBondDenom != "" {
+				defaultGenesis = mbm.DefaultGenesis(cdc)
+				stakingRaw := defaultGenesis["staking"]
+				var initialStakingData, finalStakingdata interface{}
+				err = json.Unmarshal(stakingRaw, &initialStakingData)
+				if err != nil {
+					return err
+				}
+				stakingbz, err := json.Marshal(initialStakingData)
+				if err != nil {
+					return err
+				}
```

Comment:
```suggestion
				var (
				   initialStakingData interface{}
				   finalStakingData interface{}
				 )
				 
				
				if err := json.Unmarshal(stakingRaw, &initialStakingData); err != nil {
					return err
				}
				
				bz, err := json.Marshal(initialStakingData)
				if err != nil {
					return err
				}
```

Reply:
Also, when it comes to serializing genesis state, I do not believe we should be using `json` pkg. I think we need to use proto JSON or Tendermint's json package.

Reply:
does this(https://github.com/cosmos/cosmos-sdk/pull/9776/commits/0d894c90d7574b8b39814cf3e512907347b5bdca) make sense? @alexanderbez